### ISSUE TITLE
Question about GAN Training

### ISSUE THREAD

**Issue by binnnnnny** [prajwalsingh/EEG2Image#16]
Hi, thanks for your great work ! I have some questions while applying my dataset.

1. When learning GAN for train.pyin [test_batch](https://github.com/prajwalsingh/EEG2Image/blob/aa15b985a8c80fb51af4c36e6ddc942ad883dcb8/train.py#L71), [the image is a training dataset](https://github.com/prajwalsingh/EEG2Image/blob/aa15b985a8c80fb51af4c36e6ddc942ad883dcb8/train.py#L28) but EEG signal is test data. Should I go through the inference process once more with test_generation.py after learning at train.py ?

2.  I'm trying to get the Inception Score of the paper, but the results are somehow different from the values in the paper. I got 8.04 Inception Score. 

```
1) Prepare your dataset, LSTM ckpt, GAN ckpt
2) Run test_generation.py
3) Run inceptionscore.py
```

Can you tell me what the problem is?
 
Thanks!

**Comment by prajwalsingh** [comment]
Hi @binnnnnny , thank you for reading our work.

1. The reason why test images are similar to train images is because of the way the dataset is collected. For more details, refer to this issue [_prajwalsingh/EEG2Image/issues/4_].

2. Please refer to this [_prajwalsingh/EEG2Image/issues/2_] regarding the inception score clarification.

**Comment by Cirri12** [comment]
Hi @prajwalsingh, I want to know how do we find the best checkpoint after training the GAN ? I trained the GAN network and all the 300 checkpoints are saved, but then how do I know the best checkpoint among them to be able to run the test_generation.py file. I ran with your provided GAN checkpoint and that ran successfully and I saw you got 210 checkpoint as the best one. So how did you arrive at that checkpoint?
Thanks for the great work!
